vault/barrier_aes_gcm.go

```diff
@@ -653,6 +656,7 @@ func (b *AESGCMBarrier) Put(ctx context.Context, entry *Entry) error {
 		Value:    b.encrypt(entry.Key, term, primary, entry.Value),
 		SealWrap: entry.SealWrap,
 	}
+	b.l.RUnlock()
```

Comment:
Can this go before allocating `pe`?

Reply:
_Probably_? I wasn't sure about the aead stuff but...I don't see why it can't.